src/bitcoin/wallet.rs

```diff
@@ -88,6 +90,19 @@ impl Wallet {
         Ok(Wallet { wallet })
     }
 
+    pub fn start_full_scan(&self) -> FullScanRequest {
+        self.wallet.start_full_scan().build().into()
+    }
+
+    pub fn start_sync_with_revealed_spks(&self) -> SyncRequest {
+        self.wallet.start_sync_with_revealed_spks().build().into()
+    }
+
+    pub fn apply_update_at(&mut self, update: Update, seen_at: Option<u64>) -> JsResult<()> {
+        self.wallet.apply_update_at(update, seen_at)?;
+        Ok(())
+    }
+
```

Comment:
Nit: What do you think of adding a new `apply_update_at_now` (or maybe `apply_update`?)

This method would just do what we do in the test:
```suggestion
    pub fn apply_update_at(&mut self, update: Update, seen_at: Option<u64>) -> JsResult<()> {
        self.wallet.apply_update_at(update, seen_at)?;
        Ok(())
    }

    pub fn apply_update_at_now(&mut self, update: Update) -> JsResult<()> {
        self.apply_update_at(update, Some((Date::now() / 1000.0) as u64))
    }
```

But if you're trying to have a 1:1 mapping between the BDK interface, that might not be important.

WDYT?

Reply:
Thanks for the review! Here are 2 possibilities:
- Given that `now()` is specific to the system you're in, `bdk` has an[ `apply_update`](https://docs.rs/bdk_wallet/latest/bdk_wallet/struct.Wallet.html#method.apply_update) function that is specific to `std`. `Date` does not exist by default and we need to get it from the `js-sys` crate, meaning you caught that `js-sys` is actually a dev dependency for the tests only and I can remove from `dependencies`.
- We could implement an `update_at` that uses `js-sys::Date` instead of letting the user choose what library he wants to get the `now()` from.

I guess it's opinionated vs un-opinionated? We know that `now()` always exists in any JS environment.

Reply:
As discussed together internally, using `js-sys::Date` seems like a good idea!

Reply:
Done!

Reply:
Thanks!